Wikipedia/UI-V5/WMFArticleListCollectionViewController.m

```diff
@@ -24,16 +25,28 @@
 
 
 @interface WMFArticleListCollectionViewController ()
-
+<UICollectionViewDelegate>
 @property (strong, nonatomic) MWKArticle* selectedArticle;
+@property (nonatomic, strong) IBOutlet UICollectionView* collectionView;
+@property (nonatomic, strong) IBOutlet UICollectionViewLayout* collectionViewLayout;
+
++ (Class)collectionViewClass;
 
 @end
 
 @implementation WMFArticleListCollectionViewController
 @synthesize listTransition = _listTransition;
 
-- (instancetype)init {
-    return [self initWithCollectionViewLayout:[SelfSizingWaterfallCollectionViewLayout new]];
++ (SelfSizingWaterfallCollectionViewLayout*)createLayout {
+    return [SelfSizingWaterfallCollectionViewLayout new];
+}
+
++ (Class)collectionViewClass {
+    return [UICollectionView class];
+}
+
++ (UICollectionView*)createCollectionView {
```

Comment:
Why do this instead of using the purpose built initializer?

```
- (instancetype)init {
    return [self initWithCollectionViewLayout:[SelfSizingWaterfallCollectionViewLayout new]];
}
```